querier.cc

```diff
@@ -121,7 +121,7 @@ static bool ranges_match(const schema& s, const dht::partition_range& original_r
     return bound_eq(original_range.end(), new_range.end());
 }
 
-static bool ranges_match(const schema& s, dht::partition_ranges_view original_ranges, dht::partition_ranges_view new_ranges) {
+static bool ranges_match(const schema& s, const dht::partition_ranges_view& original_ranges, const dht::partition_ranges_view& new_ranges) {
```

Comment:
partition_ranges_view is just two words (btw - can be replaced with std::span now). Passing it is cheap while dereferencing it is more expensive. I don't think this is an optimization (unless - the compiler refuses to pass it in registers).

Reply:
I'll measure perf-simple-query on this change alone.

Reply:
Hmm, it would be up to 30 instruction/op give or take a few as the next patch shows...
so I'll drop both